docs/Transport Manager Programming/index.md

```diff
@@ -1,567 +1,289 @@
-
-  * Means that the transport layer creates a physical connection for sending and receiving "handshake" data
-  * Does not mean that this application or device shall be marked as "connected" for the user
-  * May be a one time connection with which data related to the device is cleared after disconnection
-  * May be a persistent connection with which information about the device is stored in a powered off state
-
-"Connection Closing":
-
-  * Means that connection is not terminated until final 'goodbye' data transmission has completed
-  * Physical disconnection happens when there is no more data left for transmission
-
-!!!
-
-## Transport Level Structure
-
-Figure 1 demonstrates the structure of the Transport Level
-
-!!! note
-
-  * The transport manager has no limitations on the number of transports
-  * Each transport has no limitation on the number of devices
-  * Any device may be connected through any number of transports. In this case each connection established between the application on a device and the transport manager has a unique identifier
-
-!!!
+Hello, this guide will explain how transports work in SDL Core. We will highlight the responsibilities of each interface as well as look at some examples of transports already implemented in SDL Core. First let's take a look at Figure 1, a diagram showing the structure of the main transport components and then we will work our way down the diagram describing each component.
 
 |||
-Figure 1
+Figure 1: Transport Overview
 ![TAStructure](./assets/tastructure.png)
 |||
 
-### Transport Manager Structure
-
-Figure 2 represents the structure of the Transport Manager
-
-|||
-Figure 2
-![TM](./assets/tm.png)
-|||
+## Transport Manager
 
-### Transport Adapter Structure
-
-Figure 3 represents the structure of a Transport Adapter
-
-|||
-Figure 3
-![TA](./assets/ta.png)
-|||
-
-## Operation Examples
+The Transport Manager is responsible for the sending and receiving of data, as well as the connecting and disconnecting of devices. These responsibilities are abstracted away by Transport Adapters. A Transport Manager can contain any number of Transport Adapters, each of which is responsible for handling communication via one type of transport, such as TCP or Bluetooth. The Transport Manager also contains data necessary to handle its responsibilities, such as a mapping of each device to the Transport Adapter it uses to communicate. Other components within Core are also able to register a Transport Manager Listener with the manager, which will receive events from the Transport Manager. The default Transport Manager follows the singleton pattern, but this is not necessary if you would like to use a custom solution.
 
-|||
-New Device Search
-![New Device Search](./assets/newDeviceSearch.png)
-|||
+#### Transport Manager Structure
 
 |||
-New Device Connection
-![New Connection](./assets/newDeviceConnection.png)
+Figure 2: Transport Manager UML Diagram
```

Comment:
I would talk more about this diagram. Also this diagram might need updating. There is no transport manager listener implementation class for example. The protocol handler and the connection handler are the two classes that implement that interface.

Reply:
Yea the TransportAdapterListnerImplementation communicates specific events to the transport manager on behalf of the transport adapter. I dont think the transport adapter actually implements the transport adapter listener interface. 

This diagram might be wrong and is confusing. Not sure if it is showing the inheritance hierarchy, or how data is supposed to "flow" through the components.

Reply:
It's UML so it shows inheritance. I will add some basic UML information underneath in an !!!NOTE block. I didn't find any of this diagram to be incorrect, but I will take another look to make sure it's not causing more confusion than it's worth.

Reply:
Inside the UML refresher note, I added the language

> Implementation (*Impl) classes only represent concretions and may not actually be named the same in the SDL core project.

In the header above the two UML figures, I changed `* Structure` to `* Inheritance Structure`.

These two changes should clear up that this is just showing the inheritance structure.